tasks/main.yml

```diff
@@ -18,11 +18,13 @@
 
 - name: open permissions on settings file.
   file:
-    path: "{{ deploydrupal_core_path }}/sites/{{ deploydrupal_site_name }}/settings.php"
+    path: "{{ deploydrupal_core_path }}/sites/{{ deploydrupal_site_name }}/"
     state: file
     owner: "{{ deploydrupal_checkout_user }}"
     group: "{{ deploydrupal_checkout_user }}"
     mode: ug+w
+  with_fileglob:
```

Comment:
This is what I'm looking for input on. What is the best way to do this?

Reply:
So these `with_` constructs are [lookups](https://docs.ansible.com/ansible/latest/plugins/lookup.html#using-lookup-plugins). The thing about lookups is that they run _on the controlling machine_, so they're basically ruled out for this usage. I believe `with_*` is also deprecated in favour of e.g. `"{{ lookup('fileglob', '/path/to/files/**/*.ext') }}"` though still technically supported.

Also, your pattern will miss `settings.php` :smiley: 

The main available options are:

- Use the `command` module (we use `shell` elsewhere for some cli stuff because `drush` doesn't play nicely with `command`, but we should use `command` where possible):

      - name: open permissions on settings file.
        command: >
          chmod ug+w {{ deploydrupal_core_path }}/sites/{{ deploydrupal_site_name }}/{{ deploydrupal_settings_file_pattern }}

- Use a loop with file paths provided manually:

      # Assumes a variable like this is set somewhere, but defaults to `['settings.php']`:
      #
      # deploydrupal_settings_files:
      #   - settings.php
      #   - settings.environment.php
      #   - settings.local.php 
      - name: open permissions on settings file.
        file:
          path: "{{ deploydrupal_core_path }}/sites/{{ deploydrupal_site_name }}/{{ settings_file }}"
          state: file
          owner: "{{ deploydrupal_checkout_user }}"
          group: "{{ deploydrupal_checkout_user }}"
          mode: ug+w
        loop: "{{ deploydrupal_settings_files | default(['settings.php']) }}"
        loop_control:
          loop_var: settings_file

- Use a loop with _discovered_ settings files. The discovery could be done using the `find` shell command, or the `find` Ansible module. That would look like the previous task, except there'd be an additional task running before it that defines a variable used here and in the `close permissions` task below (note also that since `find` returns the path, the entire `path` in the `file` task should be replaced with `{{ settings_file }}`):

      - name: discover settings files.
        command: "find {{ deploydrupal_core_path }}/sites/{{ deploydrupal_site_name }} -type f -name 'settings*.php'"
        register: deploydrupal_settings_files

Reply:
@ctorgalson as always, thank you for you Ansible wisdom! 🧙‍♂️